src/Transport/Config/AzureServiceBusTransportInfrastructure.cs

```diff
@@ -55,6 +55,11 @@ public override string ToTransportAddress(LogicalAddress logicalAddress)
 
         public override TransportReceiveInfrastructure ConfigureReceiveInfrastructure()
         {
+            if (topology is ForwardingTopology t)
```

Comment:
This seems to me a clear indication of a responsibility problem. 

If I understand correctly the following happens here

1. FindOutHowManyTopicExistsForBundle is called in a factory that is synchronous and only supped to return the receive infrastructure factory.
1. The method is called find but actually does insert the number of topics into the newly introduced `NamespaceBundleConfigurations`
1. The bundle check is called check and is asynchronous but is called synchronous and does not really check but fill the bundle configurations
1. The bundle configurations is then used only by the ForwardingTopologySectionManager in order to fill the topics by reading again from `settings.Get<NamespaceBundleConfigurations>(WellKnownConfigurationKeys.Topology.Bundling.NamespaceBundleConfigurations);`

For me this sounds like an unnecessary complex scattering of responsibilities which arguably belongs into the section manager.